项目重新yarn install后报错 Module not found: Can't resolve '@vue/server-renderer'

<!-- 请不要删除自动生成的 Issue 标签 -->
<!-- 请不要删除自动生成的 Issue 标签 -->
### 相关平台
微信小程序

### 复现仓库
[https://github.com/linfun486/taro-2020.9.24-issue](https://github.com/linfun486/taro-2020.9.24-issue)
**小程序基础库: 2.13.0**
**使用框架: Vue 3**

### 复现步骤
taro init 的vue3-vuex 就存在此问题

步骤1: taro init 然后选 Vue3/  ts[Y] / Sass / Gitee / vue3-vuex
步骤2: taro build --type weapp --watch

就报错了，项目中也是同样的错误


### 期望结果
能正常跑起来build 

### 实际结果
生成  工具配置  /Users/linfuchuan/taro/test/dist/project.config.json

编译  发现入口  src/app.ts
编译  发现页面  src/pages/index/index.vue
🙅   编译失败.

./src/components/NumberDisplay.vue?vue&type=template&id=6aab00b9&bindings={"getNumbers":"setup"})
Module not found: Can't resolve '@vue/server-renderer' in '/Users/linfuchuan/taro/test/src/components'


监听文件修改中...

./src/components/NumberDisplay.vue?vue&type=template&id=6aab00b9&bindings={"getNumbers":"setup"} (./node_modules/@tarojs/mini-runner/node_modules/babel-loader/lib!./node_modules/vue-loader/dist/templateLoader.js??ref--6!./node_modules/vue-loader/dist??ref--10-0!./src/components/NumberDisplay.vue?vue&type=template&id=6aab00b9&bindings={"getNumbers":"setup"})
Module not found: Error: Can't resolve '@vue/server-renderer' in '/Users/linfuchuan/taro/test/src/components'
resolve '@vue/server-renderer' in '/Users/linfuchuan/taro/test/src/components'
。。。。

### 环境信息
```
Taro CLI 3.0.11 environment info:
    System:
      OS: macOS 10.15.6
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 10.22.1 - /usr/local/bin/node
      Yarn: 1.22.5 - /usr/local/bin/yarn
      npm: 6.14.6 - /usr/local/bin/npm
    npmPackages:
      @tarojs/components: 3.0.11 => 3.0.11 
      @tarojs/mini-runner: 3.0.11 => 3.0.11 
      @tarojs/runtime: 3.0.11 => 3.0.11 
      @tarojs/taro: 3.0.11 => 3.0.11 
      @tarojs/webpack-runner: 3.0.11 => 3.0.11 
      babel-preset-taro: 3.0.11 => 3.0.11 
      eslint-config-taro: 3.0.11 => 3.0.11 
```


---
### 补充信息
在2020.9.14 通过3.0.10  完成init 没有任何问题，但是昨天（2020.9.23）想打包发线上预览报错，getCurrentInstance获取到的ctx 与开发环境不一致，于是想reinstall，于是出现这个错误，尝试过3.0.9, 3.0.10, 3.0.11都是相同的情况，期间没有更改过package.json

怀疑taro相关的依赖更新后产生的问题。

<!-- generated by taro-issues. 请勿修改或删除此行注释 --><!--labels=T-weapp,V-3,F-vue3-->

--- a/packages/taro-mini-runner/src/plugins/MiniPlugin.ts
+++ b/packages/taro-mini-runner/src/plugins/MiniPlugin.ts
@@ -22,7 +22,8 @@ import {
   NODE_MODULES_REG,
   FRAMEWORK_EXT_MAP,
   printLog,
-  processTypeEnum
+  processTypeEnum,
+  FRAMEWORK_MAP
 } from '@tarojs/helper'
 
 import TaroSingleEntryDependency from '../dependencies/TaroSingleEntryDependency'
@@ -64,14 +65,19 @@ interface FilesConfig {
   }
 }
 
-export const createTarget = function createTarget (_) {
+export const createTarget = function createTarget ({ framework }) {
   return (compiler: webpack.compiler.Compiler) => {
     const { options } = compiler
     new JsonpTemplatePlugin().apply(compiler)
     new FunctionModulePlugin(options.output).apply(compiler)
     new NodeSourcePlugin(options.node).apply(compiler)
     if (process.env.NODE_ENV !== 'jest') {
-      new LoaderTargetPlugin('node').apply(compiler)
+      // 暂时性修复 vue3 兼容问题，后续再改进写法
+      if (framework === FRAMEWORK_MAP.VUE3) {
+        new LoaderTargetPlugin('web').apply(compiler)
+      } else {
+        new LoaderTargetPlugin('node').apply(compiler)
+      }
     }
   }
 }

--- a/packages/taro-mini-runner/src/webpack/build.conf.ts
+++ b/packages/taro-mini-runner/src/webpack/build.conf.ts
@@ -193,7 +193,9 @@ export default (appPath: string, mode, config: Partial<IBuildConfig>): any => {
       publicPath: '/',
       globalObject
     }, output]),
-    target: createTarget(buildAdapter),
+    target: createTarget({
+      framework
+    }),
     resolve: { alias },
     module: getModule(appPath, {
       sourceDir,

